crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -0,0 +1,312 @@
+use alloc::{collections::BTreeSet, sync::Arc, vec::Vec};
+
+use miden_objects::{
+    account::{Account, AccountId, AccountProcedureInfo},
+    assembly::mast::{MastForest, MastNode, MastNodeId},
+    crypto::dsa::rpo_falcon512,
+    note::{Note, NoteScript},
+    Digest,
+};
+
+use crate::{
+    account::components::{
+        basic_fungible_faucet_library, basic_wallet_library, rpo_falcon_512_library,
+    },
+    note::scripts::{p2id, p2idr, swap},
+    AuthScheme,
+};
+
+/// Possible variations of result whether some note be consumed by some account.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum CheckResult {
+    Yes,
+    No,
+    // `Maybe` variant means that the account has all necessary procedures to execute the note,
+    // but the correctness of the note script could not be guaranteed.
+    Maybe,
+}
```

Comment:
I'd probably try to make this more specific. Maybe something like `NoteConsumabilityResult` - but also maybe there is something better.

Also, I'd provide doc comments for all variants and change the enum doc comment to something like: "Describes whether a note can be consumed by a specific account."

Reply:
First I thought to say that I'd prefer the `NoteAccountConsumability`, but as you mentioned below even the `Yes` result can't guarantee that the note could be consumed, so it is indeed more like a compatibility than a consumability. So I agree that `NoteAccountCompatibility` is the best option for the name.